src/NuGet.CommandLine/Commands/PushCommand.cs

```diff
@@ -48,10 +61,9 @@ public class PushCommand : Command
                 timeout = TimeSpan.FromMinutes(5); // Default to 5 minutes
             }
 
-            source = await GetPushEndpointAsync(source);
-            PushPackage(packagePath, source, apiKey, timeout);
+            PushPackage(packagePath, pushEndpoint, apiKey, timeout);
 
-            if (source.Equals(NuGetConstants.DefaultGalleryServerUrl, StringComparison.OrdinalIgnoreCase))
+            if (pushEndpoint.Equals(NuGetConstants.DefaultGalleryServerUrl, StringComparison.OrdinalIgnoreCase))
```

Comment:
Apparently, this is just a legacy behavior, we never broke. I can file a bug if we want. IMO, we should not fix it for this release